Yoast/javascript
# Create word occurrences bar component

## Summary


  * Package(s) involved: components

  * Should the change be included in the package changelog?
  * Should the change be included in one or more plugin changelogs?
  * Package changelog item (if applicable): _[behind feature flag]_ Added the WordOccurrences component, which is a new visualisation of the WordList capable of handling improved analysis results.
  * Plugin changelog item (if applicable): _[behind feature flag]_ Changed the visualisation of the Prominent Words list to be more intuitive at a glance.

## Relevant technical choices:

* Using the WordCloud turned out to not be feasible due to technical shortcomings (Slack communication, product channel June 4th 2019). Therefore we changed the exterior to look like a horizontal bar-plot (see https://github.com/Yoast/wordpress-seo-premium/issues/2297#issuecomment-498749242 ).

## Test instructions

This PR can be tested by following these steps:

1. Checkout branch in javascript, start the `app/components` app, and verify that the example component works as expected.
2. Run `yarn link-monorepo` in wordpress-seo-premium (trunk), and have the plugin testing tool installed and active.
3. Have a post with a decent amount of text in it, preferably an actual text (not the same word copied 100000 times).

4. If you don't have the `improved internal linking` feature activated in the testing tool, make sure the Insights tab is functioning as expected (as in, nothing changed). Then, activate the feature flag.

5. Verify that the most occurring word has a bar behind it with the full width, and that all fewer occurrences are proportional to that bare (i.e. an occurrence of 30 should be approximately half the length of an occurrence of 60, etc.).

6. Check that the VoiceOver makes sense for the insights tab (cmd + f5 in safari)

7. If you haven't done so, verify that without the `improved internal linking` feature flag active, everything remains the same.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Insights tab, prominent words list is now revamped.

## UI changes

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo-premium/issues/2297
